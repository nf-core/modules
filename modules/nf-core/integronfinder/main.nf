process INTEGRONFINDER {
    tag "$meta.id"
    label 'process_low'

    containerOptions workflow.containerEngine == 'singularity' ? 
        '--env MPLCONFIGDIR=/tmp/mplconfigdir' :
        '-e MPLCONFIGDIR=/tmp/mplconfigdir'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/integron_finder:2.0.5--pyhdfd78af_0':
        'biocontainers/integron_finder:2.0.5--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*/*.gbk")              , emit: gbk, optional: true
    tuple val(meta), path("*/*.integrons")        , emit: integrons
    tuple val(meta), path("*/*.summary")          , emit: summary
    tuple val(meta), path("*/integron_finder.out"), emit: out
    path "versions.yml"                           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args                = task.ext.args ?: ''
    def is_compressed_fasta = fasta.getName().endsWith(".gz") ? true : false
    def fasta_name          = fasta.getName().replace(".gz", "")
    
    """
    if [ "$is_compressed_fasta" == "true" ]; then
        gzip -c -d $fasta > $fasta_name
    fi

    integron_finder \\
        $args \\
        --cpu $task.cpus \\
        $fasta_name

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        integronfinder: \$(integron_finder --version 2>&1 | head -n1 | sed 's/^integron_finder version //' | cut -d ' ' -f1)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def gbk_create = args.contains("--gbk") ? "Results_Integron_Finder_${prefix}/${prefix}.gbk" : ""

    """
    mkdir -p Results_Integron_Finder_${prefix}
    ${gbk_create}
    touch "Results_Integron_Finder_${prefix}/${prefix}.integrons"
    touch "Results_Integron_Finder_${prefix}/${prefix}.summary"
    touch "Results_Integron_Finder_${prefix}/integron_finder.out"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        integronfinder: \$(integron_finder --version 2>&1 | head -n1 | sed 's/^integron_finder version //' | cut -d ' ' -f1)
    END_VERSIONS
    """
}