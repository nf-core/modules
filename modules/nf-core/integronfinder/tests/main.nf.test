nextflow_process {

    name "Test Process INTEGRONFINDER"
    script "../main.nf"
    process "INTEGRONFINDER"

    tag "modules"
    tag "modules_nfcore"
    tag "integronfinder"

    test("IntegronFinder - escherichia_coli - genome_fa_gz") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/escherichia_coli/genome/genome.fa.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.out[0][1]).readLines().get(56).contains("command used: integron_finder") },
                { assert snapshot(
                        process.out.gbk ? path(process.out.gbk[0][1]).text : "no_gbk",
                        path(process.out.summary[0][1]).readLines()[1..-1], // Remove first line due to hash presence in conda
                        path(process.out.integrons[0][1]).readLines()[1..-1],
                        process.out.versions,
                        path(process.out.versions[0]).yaml
                    ).match() }
            )
        }

    }

    test("IntegronFinder - escherichia_coli - genome_fa_gz - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/escherichia_coli/genome/genome.fa.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
