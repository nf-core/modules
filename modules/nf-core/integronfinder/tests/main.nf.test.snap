{
    "IntegronFinder - escherichia_coli - genome_fa_gz": {
        "content": [
            "no_gbk",
            [
                "# cmd: integron_finder --cpu 2 genome.fa",
                "ID_replicon\tCALIN\tcomplete\tIn0\ttopology\tsize",
                "chr\t0\t0\t0\tcirc\t48502"
            ],
            [
                "# cmd: integron_finder --cpu 2 genome.fa",
                "# No Integron found"
            ],
            [
                "versions.yml:md5,381190cf27737ed39ada1dfeec09dbe8"
            ],
            {
                "INTEGRONFINDER": {
                    "integronfinder": "2.0.5"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-02T18:53:05.832325584"
    },
    "IntegronFinder - escherichia_coli - genome_fa_gz - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.integrons:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "integron_finder.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,381190cf27737ed39ada1dfeec09dbe8"
                ],
                "gbk": [
                    
                ],
                "integrons": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.integrons:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "integron_finder.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.summary:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,381190cf27737ed39ada1dfeec09dbe8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-02T16:19:41.207197789"
    }
}